### PR TITLE
Fix some inconsistencies in pmix_status_t vs int declarations

### PR DESCRIPTION
**Pull request by jjhursey**
* These were found when building on power7 with XL compiler
 * See the following PR from Open MPI for more details:
   - https://github.com/open-mpi/ompi-release/pull/1129

### CODE REVIEW

**Comment by jjhursey** [comment]
:bot:assign: @rhc54
:bot:label:bug
:bot:milestone:v1.1.4

File: src/buffer_ops/internal.h [hunk 1]
```diff
@@ -87,8 +88,8 @@ BEGIN_C_DECLS
 #endif
 
 /*
- * Internal type corresponding to int and unsigned int.  Do not use
- * this in interface calls - use PMIX_INT / PMIX_UINT instead.
+ * Internal type corresponding to pmix_status_t and unsigned int.  Do not use
```
Comment (rhc54) [inline src/buffer_ops/internal.h]: oops - think this should remain int

File: src/buffer_ops/internal.h [hunk 2]
```diff
@@ -87,8 +88,8 @@ BEGIN_C_DECLS
...
+ * this in interface calls - use PMIX_pmix_status_t / PMIX_Upmix_status_t instead.
```
Comment (rhc54) [inline src/buffer_ops/internal.h]: oops again

File: src/buffer_ops/internal.h [hunk 3]
```diff
@@ -185,7 +186,7 @@ typedef struct {
     pmix_bfrop_unpack_fn_t odti_unpack_fn;
     /** copy function */
     pmix_bfrop_copy_fn_t odti_copy_fn;
-    /** print function */
+    /** prpmix_status_t function */
```
Comment (rhc54) [inline src/buffer_ops/internal.h]: oops

File: src/buffer_ops/internal.h [hunk 4]
```diff
@@ -244,214 +245,214 @@ PMIX_DECLSPEC    pmix_status_t pmix_bfrop_unpack_buffer(pmix_buffer_t *buffer, v
...
-int pmix_bfrop_copy_bo(pmix_byte_object_t **dest, pmix_byte_object_t *src,
+pmix_status_t pmix_bfrop_copy_bo(pmix_byte_object_t **dest, pmix_byte_object_t *src,
                        pmix_data_type_t type);
-int pmix_bfrop_copy_pdata(pmix_pdata_t **dest, pmix_pdata_t *src,
+pmix_status_t pmix_bfrop_copy_pdata(pmix_pdata_t **dest, pmix_pdata_t *src,
                           pmix_data_type_t type);
 
 /*
- * Internal print functions
+ * Internal prpmix_status_t functions
```
Comment (rhc54) [inline src/buffer_ops/internal.h]: oops

File: src/buffer_ops/print.c [hunk 5]
```diff
@@ -41,7 +41,7 @@ int pmix_bfrop_print(char **output, char *prefix, void *src, pmix_data_type_t ty
         return PMIX_ERR_BAD_PARAM;
     }
 
-    /* Lookup the print function for this type and call it */
+    /* Lookup the prpmix_status_t function for this type and call it */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 6]
```diff
@@ -51,17 +51,17 @@ int pmix_bfrop_print(char **output, char *prefix, void *src, pmix_data_type_t ty
 }
 
 /*
- * STANDARD PRINT FUNCTIONS FOR SYSTEM TYPES
+ * STANDARD PRpmix_status_t FUNCTIONS FOR SYSTEM TYPES
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 7]
```diff
@@ -51,17 +51,17 @@ int pmix_bfrop_print(char **output, char *prefix, void *src, pmix_data_type_t ty
...
+pmix_status_t pmix_bfrop_print_bool(char **output, char *prefix, bool *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 8]
```diff
@@ -79,15 +79,15 @@ int pmix_bfrop_print_bool(char **output, char *prefix, bool *src, pmix_data_type
...
+pmix_status_t pmix_bfrop_print_byte(char **output, char *prefix, uint8_t *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 9]
```diff
@@ -104,15 +104,15 @@ int pmix_bfrop_print_byte(char **output, char *prefix, uint8_t *src, pmix_data_t
...
+pmix_status_t pmix_bfrop_print_string(char **output, char *prefix, char *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 10]
```diff
@@ -129,15 +129,15 @@ int pmix_bfrop_print_string(char **output, char *prefix, char *src, pmix_data_ty
...
+pmix_status_t pmix_bfrop_print_size(char **output, char *prefix, size_t *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 11]
```diff
@@ -154,15 +154,15 @@ int pmix_bfrop_print_size(char **output, char *prefix, size_t *src, pmix_data_ty
...
+pmix_status_t pmix_bfrop_print_pid(char **output, char *prefix, pid_t *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 12]
```diff
@@ -178,15 +178,15 @@ int pmix_bfrop_print_pid(char **output, char *prefix, pid_t *src, pmix_data_type
...
+pmix_status_t pmix_bfrop_print_int(char **output, char *prefix, int *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 13]
```diff
@@ -203,15 +203,15 @@ int pmix_bfrop_print_int(char **output, char *prefix, int *src, pmix_data_type_t
...
+pmix_status_t pmix_bfrop_print_uint(char **output, char *prefix, uint *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 14]
```diff
@@ -228,15 +228,15 @@ int pmix_bfrop_print_uint(char **output, char *prefix, int *src, pmix_data_type_
...
+pmix_status_t pmix_bfrop_print_uint8(char **output, char *prefix, uint8_t *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 15]
```diff
@@ -253,15 +253,15 @@ int pmix_bfrop_print_uint8(char **output, char *prefix, uint8_t *src, pmix_data_
...
+pmix_status_t pmix_bfrop_print_uint16(char **output, char *prefix, uint16_t *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 16]
```diff
@@ -287,7 +287,7 @@ int pmix_bfrop_print_uint32(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 17]
```diff
@@ -313,7 +313,7 @@ int pmix_bfrop_print_int8(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 18]
```diff
@@ -339,7 +339,7 @@ int pmix_bfrop_print_int16(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 19]
```diff
@@ -356,15 +356,15 @@ int pmix_bfrop_print_int16(char **output, char *prefix,
...
+pmix_status_t pmix_bfrop_print_int32(char **output, char *prefix, int32_t *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 20]
```diff
@@ -390,7 +390,7 @@ int pmix_bfrop_print_uint64(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 21]
```diff
@@ -417,7 +417,7 @@ int pmix_bfrop_print_int64(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 22]
```diff
@@ -443,7 +443,7 @@ int pmix_bfrop_print_float(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 23]
```diff
@@ -469,7 +469,7 @@ int pmix_bfrop_print_double(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 24]
```diff
@@ -496,7 +496,7 @@ int pmix_bfrop_print_time(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 25]
```diff
@@ -525,7 +525,7 @@ int pmix_bfrop_print_timeval(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 26]
```diff
@@ -543,12 +543,12 @@ int pmix_bfrop_print_timeval(char **output, char *prefix,
     return PMIX_SUCCESS;
 }
 
-/* PRINT FUNCTIONS FOR GENERIC PMIX TYPES */
+/* PRpmix_status_t FUNCTIONS FOR GENERIC PMIX TYPES */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 27]
```diff
@@ -557,7 +557,7 @@ int pmix_bfrop_print_value(char **output, char *prefix,
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 28]
```diff
@@ -741,19 +741,19 @@ static void print_hwloc_obj(char **output, char *prefix,
     unsigned i;
     struct hwloc_topology_support *support;
 
-    /* print the object type */
+    /* prpmix_status_t the object type */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 29]
```diff
@@ -741,19 +741,19 @@ static void print_hwloc_obj(char **output, char *prefix,
...
     hwloc_obj_type_snprintf(string, 1024, obj, 1);
     asprintf(&pfx, "\n%s\t", (NULL == prefix) ? "" : prefix);
     asprintf(&tmp, "%sType: %s Number of child objects: %u%sName=%s",
              (NULL == prefix) ? "" : prefix, string, obj->arity,
              pfx, (NULL == obj->name) ? "NULL" : obj->name);
     if (0 < hwloc_obj_attr_snprintf(string, 1024, obj, pfx, 1)) {
-        /* print the attributes */
+        /* prpmix_status_t the attributes */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 30]
```diff
@@ -741,19 +741,19 @@ static void print_hwloc_obj(char **output, char *prefix,
...
         asprintf(&tmp2, "%s%s%s", tmp, pfx, string);
         free(tmp);
         tmp = tmp2;
     }
-    /* print the cpusets - apparently, some new HWLOC types don't
+    /* prpmix_status_t the cpusets - apparently, some new HWLOC types don't
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 31]
```diff
@@ -794,7 +794,7 @@ static void print_hwloc_obj(char **output, char *prefix,
     asprintf(&pfx, "%s\t", (NULL == prefix) ? "" : prefix);
     for (i=0; i < obj->arity; i++) {
         obj2 = obj->children[i];
-        /* print the object */
+        /* prpmix_status_t the object */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 32]
```diff
@@ -804,37 +804,37 @@ static void print_hwloc_obj(char **output, char *prefix,
...
+pmix_status_t pmix_bfrop_print_topo(char **output, char *prefix,
                           hwloc_topology_t src, pmix_data_type_t type)
 {
     hwloc_obj_t obj;
     char *tmp=NULL;
 
     /* get root object */
     obj = hwloc_get_root_obj(src);
-    /* print it */
+    /* prpmix_status_t it */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 33]
```diff
@@ -804,37 +804,37 @@ static void print_hwloc_obj(char **output, char *prefix,
...
+pmix_status_t pmix_bfrop_print_persist(char **output, char *prefix, pmix_persistence_t *src, pmix_data_type_t type)
 {
     char *prefx;
 
     /* deal with NULL prefix */
     if (NULL == prefix) asprintf(&prefx, " ");
     else prefx = prefix;
 
-    /* if src is NULL, just print data type and return */
+    /* if src is NULL, just prpmix_status_t data type and return */
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

File: src/buffer_ops/print.c [hunk 34]
```diff
@@ -860,7 +860,7 @@ int pmix_bfrop_print_bo(char **output, char *prefix,
...
```
Comment (rhc54) [inline src/buffer_ops/print.c]: oops

**Comment by jjhursey** [comment]
Let me look into some of those merge issues. I tried to use the patch from Open MPI to get started, but I think it messed up these comments.

File: src/buffer_ops/internal.h (hunk 2 above)
Comment (jjhursey) [inline src/buffer_ops/internal.h]: I see this in the ompi/ompi-release v2.x branch:
  * https://github.com/open-mpi/ompi-release/blob/v2.x/opal/mca/pmix/pmix112/pmix/src/buffer_ops/internal.h#L91

I'll work up a patch for PMIx then see if I can apply it over these to fix the comments.

**Comment by jjhursey** [comment]
@rhc54 I fixed those comments. They will need to be fixed in Open MPI as well (I'll work on a patch since i have it handy).
I took a scan of the patch and I think it is clean now. Might be worth another quick scan.

**Comment by rhc54** [comment]
Looks good - thanks!

**Pull request merged**